File: nif_reader_3/src/btpy/src/btpy/btpy_checkers/mod/is_rgb/is_rgb.py


from ..is_byte.is_byte import*


def is_rgb(data)->bool:
    """
    Checker que verifica si un 
    dato es de tipo rgb lista
    """
    if(not type(data) == list): return False
    if(not len(data) == 3): return False
    for e in data:
        if(not is_byte256(e)): return False
    return True

def is_rgba(data)->bool:
    """
    Checker que verifica si un 
    dato es de tipo rgba lista
    """
    if(not type(data) == list): return False
    if(not len(data) == 4): return False
    for e in data:
        if(not is_byte256(e)): return False
    return True

File: btjs_page/btjs/mod/checkers/mod/is_rgb.js


import { is_byte_256 } from "./is_byte_256.js";

export function is_RGB(data) {
    if( ! Array.isArray(data)) {return false;}
    if( ! (data.length == 3)) {return false;}
    for(let i in data) {
        if( ! is_byte_256(data[i])) {
            return false;
        }
    }
    return true
}

export function is_RGBA(data) {
    if( ! Array.isArray(data)) {return false;}
    if( ! (data.length == 4)) {return false;}
    for(let i in data) {
        if( ! is_byte_256(data[i])) {
            return false;
        }
    }
    return true
}